src/dtls/network.c

```diff
@@ -649,6 +645,24 @@ static void FreeConnection(DPS_NetConnection* cn)
     if (cn->peerAddr) {
         DPS_DestroyAddress(cn->peerAddr);
     }
+
+    if (cn->netCtx) {
+        DPS_NetConnection* next = cn->next;
+        if (cn->netCtx->cns == cn) {
+            cn->netCtx->cns = next;
+        } else if (cn->netCtx->cns) {
+            DPS_NetConnection* prev = cn->netCtx->cns;
+            while (prev->next != cn) {
+                prev = prev->next;
+                assert(prev);
+            }
+            prev->next = next;
+        }
+    }
+    if ((cn->netCtx->state == NET_STOPPING) && !cn->netCtx->cns) {
```

Comment:
I am not clear what the check for cns is for here. cn get's freed either way below, might this result in a dangling pointer? A comment or two would be helpful.

Reply:
Added some comments.  The situation is that NetStop may be called while connections are still active or shutting down so we need to delay the closing of the rxSocket from NetStop until here when all the connections are done using it.